ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -461,6 +469,7 @@ loadSessionWithOptions recorder SessionLoadingOptions{..} dir = do
   runningCradle <- newVar dummyAs :: IO (Var (Async (IdeResult HscEnvEq,[FilePath])))
 
   return $ do
+    clientConfig <- getClientConfigAction
```

Comment:
I'm not sure this actually works? This will get the client config at _this_ point, but we consult it inside `consultCradle` which is an IO function that maybe runs later? So might we just always see a stale version of the config? I think we might need to get the config in IO inside `consultCradle`.

Reply:
It does work, in combination with invalidating the `GhcSessionIO` rule.